packages/react/roving-focus/src/RovingFocusGroup.tsx

```diff
@@ -1,144 +1,237 @@
-    prop: props.reachable,
-    defaultProp: props.defaultReachable,
-    onChange: props.onReachableChange,
+const RovingFocusGroup = React.forwardRef((props, forwardedRef) => {
+  const {
+    as = GROUP_DEFAULT_TAG,
+    orientation,
+    dir = 'ltr',
+    loop = false,
+    currentTabStopId: currentTabStopIdProp,
+    defaultCurrentTabStopId,
+    onCurrentTabStopIdChange,
+    onEntryFocus,
+    ...groupProps
+  } = props;
+  const ref = React.useRef<React.ElementRef<typeof RovingFocusGroup>>(null);
+  const composedRefs = useComposedRefs(forwardedRef, ref);
+  const itemMap = React.useRef<RovingContextValue['itemMap']>(new Map()).current;
+  const [currentTabStopId = null, setCurrentTabStopId] = useControllableState({
+    prop: currentTabStopIdProp,
+    defaultProp: defaultCurrentTabStopId,
+    onChange: onCurrentTabStopIdChange,
   });
-  const [tabStopId, setTabStopId] = React.useState<string | null>(null);
-  const itemMapRef = React.useRef<Map<string, boolean>>(new Map());
+  const [hasInteracted, setHasInteracted] = React.useState(false);
+  const [isTabbingBackOut, setIsTabbingBackOut] = React.useState(false);
+  const handleEntryFocus = useCallbackRef(onEntryFocus);
 
   React.useEffect(() => {
-    if (reachable) {
-      const itemIds = Array.from(itemMapRef.current.keys());
-      const firstActiveId = itemIds.find((id) => itemMapRef.current.get(id) === true);
-      const firstItemId = itemIds[0];
-      setTabStopId(firstActiveId || firstItemId || null);
-    } else {
-      setTabStopId(null);
+    const node = ref.current;
+    if (node) {
+      node.addEventListener(ENTRY_FOCUS, handleEntryFocus);
+      return () => node.removeEventListener(ENTRY_FOCUS, handleEntryFocus);
     }
-  }, [reachable]);
+  }, [handleEntryFocus]);
 
   return (
     <RovingFocusProvider
-      groupId={useId()}
       orientation={orientation}
       dir={dir}
       loop={loop}
-      tabStopId={tabStopId}
-      onTabStopIdChange={setTabStopId}
-      reachable={reachable}
-      onReachableChange={setReachable}
-      itemMap={itemMapRef.current}
+      itemMap={itemMap}
+      currentTabStopId={currentTabStopId}
+      onInteract={React.useCallback(() => setHasInteracted(true), [])}
+      onItemFocus={React.useCallback((tabStopId) => setCurrentTabStopId(tabStopId), [
+        setCurrentTabStopId,
+      ])}
+      onItemShiftTab={React.useCallback(() => setIsTabbingBackOut(true), [])}
     >
-      {children}
+      <Primitive
+        tabIndex={isTabbingBackOut ? -1 : 0}
+        aria-orientation={orientation}
+        data-orientation={orientation}
+        {...groupProps}
+        as={as}
+        ref={composedRefs}
+        onFocus={composeEventHandlers(props.onFocus, (event) => {
+          if (!isTabbingBackOut && event.target === event.currentTarget) {
+            const entryFocusEvent = new Event(ENTRY_FOCUS, EVENT_OPTIONS);
+            event.currentTarget.dispatchEvent(entryFocusEvent);
```

Comment:
Like I said, not a huge fan of the name here, would gladly take other suggestions :)